src/praisonai-agents/praisonaiagents/agent/agent.py

```
@@ -551,6 +551,7 @@ def __init__(
         parallel_tool_calls: bool = False,  # Gap 2: Enable parallel execution of batched LLM tool calls
         learn: Optional[Union[bool, str, Dict[str, Any], 'LearnConfig']] = None,  # Continuous learning (peer to memory)
         backend: Optional[Any] = None,  # External managed agent backend (e.g., ManagedAgentIntegration)
+        interrupt_controller: Optional['InterruptController'] = None,  # G2: Cooperative cancellation
```

Ruff (0.15.10)

[error] Undefined name `InterruptController`

(F821)